Application/Frontend/components/FriendsColumn/FriendsTab.tsx

```diff
@@ -17,7 +20,98 @@ import { IconSearch, IconArrowRight } from '@tabler/icons-react';
+
+    // State hooks for form data and validation errors
+    const [formData, setFormData] = useState<{userName: string, password: string}>({
+        userName: "user1",
+        password: "user1Pass",
+      });
+    
+      // To facilitate client-side validation
+      const [usernameError, setUsernameError] = useState(''); // The message for an incorrect username
+      const [passwordError, setPasswordError] = useState(''); // The message for a correct username but incorrect password
+
+  
+      // Redirect you to the Modifying webpage
+      function onModify() {
+        router.push('create-account');
+      }
+    
+      /**
+       * Handles form submission, sending the login request to the server and processing
+       * the response to either proceed to the application or show error messages.
+       *
+       * @param event - The form submission event
+       */
+      const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+    
+        const response = await fetch('/api/login', { // Establishing a promise
+          method: 'POST', // As we are dealing with authentication, this is the most appropriate method
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(formData), // The data that is being sent
+        });
+    
+        const data = await response.json(); // Awaiting the resolution of the promise
+    
+        if (response.ok) { // If the response was successful
+          router.push('/accord'); // Redirect the user to the main application page
+        } else {
+          if (data.error === 'Invalid username') {
+            setUsernameError('Invalid username. Please try again.'); // Client-side validation
+          } else if (data.error === 'Invalid password') {
+            setPasswordError('Invalid password. Please try again.'); // Client-side validation
+          }
+        }
+      };
+  
+      /**
+       * Updates form data state on user input.
+       *
+       * @param evt - The input change event
+       */
+      const handleChange = (evt: any) => { // Upon changes, we update the form data
+        const { name, value } = evt.target;
+        setFormData(prevData => ({ ...prevData, [name]: value }));
+      };
+
+    
+    
+    useEffect(() => { 
+        const fetchData = async () => { //
+          try {
+            const response = await fetch('/api/FriendsTab',{ // Establishing a promise, replace userSettings with your API Endpoint file
+              method: 'POST', 
+              headers: {
+                'Content-Type': 'application/json',
+              },
+              body: JSON.stringify(formData), 
+            }) // The data that is being sent);
+            if (response.ok) {  // If the Endpoint API code successfully ran, proceed 
+              const data = await response.json(); // 
+              setFriendList(data.friendList); // Set data accordingly
```

Comment:
Great use of the `Effect` hook to automatically pull data every time this component is mounted. Awesome work!